mayastor/src/grpc/controller_grpc.rs

```diff
@@ -1,132 +1,70 @@
-            bytes_unmapped: b.bytes_unmapped,
-        }
-    }
-}
-
-pub async fn controller_stats() -> GrpcResult<rpc::StatNvmeControllersReply> {
-    let rx = rpc_submit::<_, _, nexus::Error>(async move {
-        let mut res: Vec<rpc::NvmeControllerStats> = Vec::new();
-        let controllers = NVME_CONTROLLERS.controllers();
-        for name in controllers.iter() {
-            if let Some(ctrlr) = NVME_CONTROLLERS.lookup_by_name(name) {
-                let (s, r) =
-                    oneshot::channel::<Result<BlockDeviceIoStats, CoreError>>();
-
-                {
-                    let ctrlr = ctrlr.lock();
-
-                    if let Err(e) = ctrlr.get_io_stats(
-                        |stats, ch| {
-                            done_cb(ch, stats);
-                        },
-                        cb_arg(s),
-                    ) {
-                        error!(
-                            "{}: failed to get stats for NVMe controller: {:?}",
-                            name, e
-                        );
-                        continue;
-                    }
-                }
-
-                let stats = r.await.expect("Failed awaiting at io_stats");
-                if stats.is_err() {
-                    error!("{}: failed to get stats for NVMe controller", name);
-                } else {
-                    res.push(rpc::NvmeControllerStats {
-                        name: name.to_string(),
-                        stats: stats.ok().map(rpc::NvmeControllerIoStats::from),
-                    });
-                }
-            }
-        }
-
-        Ok(rpc::StatNvmeControllersReply {
-            controllers: res,
+        r.await.expect("Failed awaiting at io_stats")
+    } else {
+        Err(CoreError::BdevNotFound {
+            name: controller_name.to_string(),
         })
-    })?;
-
-    rx.await
-        .map_err(|_| Status::cancelled("cancelled"))?
-        .map_err(Status::from)
-        .map(Response::new)
+    }
 }
 
-pub async fn list_controllers() -> GrpcResult<rpc::ListNvmeControllersReply> {
-    let rx = rpc_submit::<_, _, nexus::Error>(async move {
-        let controllers = NVME_CONTROLLERS
-            .controllers()
-            .iter()
-            .filter_map(|n| {
-                NVME_CONTROLLERS
-                    .lookup_by_name(n)
-                    .map(|c| c.lock().to_grpc())
-            })
-            .collect::<Vec<_>>();
-
-        Ok(rpc::ListNvmeControllersReply {
-            controllers,
+pub async fn list_controllers() -> Vec<NvmeControllerInfo> {
+    NVME_CONTROLLERS
+        .controllers()
+        .iter()
+        .filter_map(|n| {
+            NVME_CONTROLLERS
+                .lookup_by_name(n)
+                .map(|c| c.lock().to_info())
         })
-    })?;
-
-    rx.await
-        .map_err(|_| Status::cancelled("cancelled"))?
-        .map_err(Status::from)
-        .map(Response::new)
+        .collect::<Vec<_>>()
```

Comment:
This is very confusing to read.. maybe `NVME_CONTROLLERS` should implement Iter or something along those lines... but the code was already like that so it's fine.